node/test/Test/Pos/Client/Txp/UtilSpec.hs

```diff
@@ -51,17 +52,32 @@ spec = giveGtConf $ giveNodeConf $ giveInfraConf $ giveUpdateConf $ giveCoreConf
     describe "Client.Txp.Util" $ do
         describe "createMTx" $ createMTxSpec
 
+-- GHC doesn't support impredicative polymorphism so we need a wrapper
+-- for the list below to typecheck.
+data TestFunctionWrapper
+    = forall prop. (Testable prop) => TestFunctionWrapper (InputSelectionPolicy -> prop)
+
 createMTxSpec :: HasTxpConfigurations => Spec
 createMTxSpec = do
-    prop createMTxWorksWhenWeAreRichDesc createMTxWorksWhenWeAreRichSpec
-    prop stabilizationDoesNotFailDesc stabilizationDoesNotFailSpec
-    prop feeIsNonzeroDesc feeIsNonzeroSpec
-    prop manyUtxoTo1Desc manyUtxoTo1Spec
-    prop manyAddressesTo1Desc manyAddressesTo1Spec
-    prop manyAddressesToManyDesc manyAddressesToManySpec
+    let inputSelectionPolicies =
+            [("Grouped inputs", OptimizeForSecurity)
+            ,("Ungrouped inputs", OptimizeForSize)
+            ]
+    let testSpecs =
+            [(createMTxWorksWhenWeAreRichDesc, TestFunctionWrapper createMTxWorksWhenWeAreRichSpec)
+            ,(stabilizationDoesNotFailDesc, TestFunctionWrapper stabilizationDoesNotFailSpec)
+            ,(feeIsNonzeroDesc, TestFunctionWrapper feeIsNonzeroSpec)
+            ,(manyUtxoTo1Desc, TestFunctionWrapper manyUtxoTo1Spec)
+            ,(manyAddressesTo1Desc, TestFunctionWrapper manyAddressesTo1Spec)
+            ,(manyAddressesToManyDesc, TestFunctionWrapper manyAddressesToManySpec)
+            ,(txWithRedeemOutputFailsDesc, TestFunctionWrapper txWithRedeemOutputFailsSpec)
+            ,(feeForManyAddressesDesc, TestFunctionWrapper feeForManyAddressesSpec)
+            ]
+    sequence_ $ do
+        (funcDesc, TestFunctionWrapper func) <- testSpecs
+        (inputSelectionDesc, inputSelectionPolicy) <- inputSelectionPolicies
+        return $ prop (inputSelectionDesc <> ": " <> funcDesc) (func inputSelectionPolicy)
```

Comment:
I think it's better to do `prop inputSelectionDesc . prop funcDesc` to get true hierarchy